sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpCompilationHelper.cs

```diff
@@ -23,9 +23,11 @@
 
 namespace SonarAnalyzer.Helpers
 {
-    public static class CSharpCompilationHelper
+    internal static class CSharpCompilationHelper
     {
-        public static bool IsAtLeastLanguageVersion(this Compilation compilation, LanguageVersion languageVersion) =>
-            (compilation as CSharpCompilation)?.LanguageVersion.CompareTo(languageVersion) >= 0;
+        internal static bool IsAtLeastLanguageVersion(this Compilation compilation, LanguageVersion languageVersion) =>
+            compilation.GetLanguageVersion().CompareTo(languageVersion) >= 0;
```

Comment:
You could reuse the `IsAtLeast` extension method you declared in `LanguageVersionExtensions` here